src/main/java/org/folio/dew/batch/acquisitions/edifact/jobs/ExportHistoryTasklet.java

```diff
@@ -30,26 +32,40 @@ public class ExportHistoryTasklet implements Tasklet {
 
   private final KafkaService kafkaService;
   private final ObjectMapper objectMapper;
+  private final OrganizationsService organizationsService;
+
   @Value("#{jobParameters['jobId']}")
   private String jobId;
   @Override
-  public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) {
+  public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) throws Exception {
     var exportHistory = buildExportHistory(chunkContext);
 
     kafkaService.send(KafkaService.Topic.EXPORT_HISTORY_CREATE, null, exportHistory);
 
     return RepeatStatus.FINISHED;
   }
 
-  ExportHistory buildExportHistory(ChunkContext chunkContext) {
+  ExportHistory buildExportHistory(ChunkContext chunkContext) throws Exception {
+    var jobParameters = chunkContext.getStepContext().getJobParameters();
+    var ediExportConfig = objectMapper.readValue((String)jobParameters.get("edifactOrdersExport"), VendorEdiOrdersExportConfig.class);
+    var vendorId = ediExportConfig.getVendorId().toString();
+    var vendor = organizationsService.getOrganizationById(vendorId);
+    var vendorName = vendor.get("code").asText();
     var stepExecutionContext = chunkContext.getStepContext().getStepExecution();
     var polineIds = getPoLineIdsFromExecutionContext(stepExecutionContext);
+    var fileName = ExecutionContextUtils.getExecutionVariable(stepExecutionContext, "edifactFileName");
+    var jobStatus = chunkContext.getStepContext().getStepExecution().getJobExecution().getStatus().toString();
```

Comment:
Did you check if for example job finished with some exception, is job execution status will be failed in this case?

Reply:
yes , i have tested with some failed cases and process failed to previous step only. control is not moving to the last step(createExportHistory).

Reply:
so fo now we will send only jobs with Succesful status